src/test/java/ShopperCartAnnotationWithExtensionTest.java

```diff
@@ -0,0 +1,28 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class ShopperCartAnnotationWithExtensionTest {
```

Comment:
this annotation takes care not only about mock initialization :)

there are also cases like "unnecessary stubbing causes failure"

Reply:
how about aso adding and example of inject-mocks?